### PR TITLE
EWPP-658: Add Consultation content type.

### PR DESCRIPTION
**Pull request by 22Alexandra**

### CODE REVIEW

File: modules/oe_content_consultation/config/install/core.base_field_override.node.oe_consultation.title.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  config:
+    - node.type.oe_consultation
+id: node.oe_consultation.title
+field_name: title
+entity_type: node
+bundle: oe_consultation
+label: Page title
+description: 'The ideal length is 50 to 60 characters including spaces. If it must be longer, make sure you fill in a shorter version in the Alternative title field.'
+required: true
+translatable: true
```
Comment (yenyasinn) [inline modules/oe_content_consultation/config/install/core.base_field_override.node.oe_consultation.title.yml:13]: `translatable: false`. See https://github.com/openeuropa/oe_content/pull/254/files

File: modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_consultation_aim.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  module:
+    - text
+id: node.oe_consultation.oe_consultation_aim
+field_name: oe_consultation_aim
+entity_type: node
+bundle: oe_consultation
+label: 'Why we are consulting'
+description: ''
+required: false
+translatable: true
```
Comment (yenyasinn) [inline modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_consultation_aim.yml:16]: `translatable: false`

File: modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_consultation_closed_text.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  module:
+    - text
+id: node.oe_consultation.oe_consultation_closed_text
+field_name: oe_consultation_closed_text
+entity_type: node
+bundle: oe_consultation
+label: 'Respond to the consultation (closed status text)'
+description: 'Specify the text that is displayed once the consultation is closed.'
+required: false
+translatable: true
```
Comment (yenyasinn) [inline modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_consultation_closed_text.yml:16]: `translatable: false`

File: modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_consultation_guidelines.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  module:
+    - text
+id: node.oe_consultation.oe_consultation_guidelines
+field_name: oe_consultation_guidelines
+entity_type: node
+bundle: oe_consultation
+label: 'Respond to the consultation'
+description: 'Specify how to submit the response and how to view the questionnaire of the consultation.'
+required: false
+translatable: true
```
Comment (yenyasinn) [inline modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_consultation_guidelines.yml:16]: `translatable: false`

File: modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_consultation_legal_info.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  module:
+    - text
+id: node.oe_consultation.oe_consultation_legal_info
+field_name: oe_consultation_legal_info
+entity_type: node
+bundle: oe_consultation
+label: 'Legal notice'
+description: ''
+required: false
+translatable: true
```
Comment (yenyasinn) [inline modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_consultation_legal_info.yml:16]: `translatable: false`

File: modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_consultation_outcome.yml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  module:
+    - text
+id: node.oe_consultation.oe_consultation_outcome
+field_name: oe_consultation_outcome
+entity_type: node
+bundle: oe_consultation
+label: 'Consultation outcome'
+description: ''
+required: false
+translatable: true
```
Comment (yenyasinn) [inline modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_consultation_outcome.yml:16]: `translatable: false`

File: modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_consultation_response_button.yml [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  module:
+    - link
+id: node.oe_consultation.oe_consultation_response_button
+field_name: oe_consultation_response_button
+entity_type: node
+bundle: oe_consultation
+label: 'Respond button'
+description: ''
+required: false
+translatable: true
```
Comment (yenyasinn) [inline modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_consultation_response_button.yml:16]: `translatable: false`

File: modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_consultation_target_audience.yml [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+  module:
+    - text
+id: node.oe_consultation.oe_consultation_target_audience
+field_name: oe_consultation_target_audience
+entity_type: node
+bundle: oe_consultation
+label: 'Target audience'
+description: 'Specify who should contribute to this consultation.'
+required: true
+translatable: true
```
Comment (yenyasinn) [inline modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_consultation_target_audience.yml:16]: `translatable: false`

File: modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_departments.yml [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+  module:
+    - rdf_skos
+id: node.oe_consultation.oe_departments
+field_name: oe_departments
+entity_type: node
+bundle: oe_consultation
+label: Departments
+description: ''
+required: false
+translatable: true
```
Comment (yenyasinn) [inline modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_departments.yml:16]: `translatable: false`

File: modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_subject.yml [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+  module:
+    - rdf_skos
+id: node.oe_consultation.oe_subject
+field_name: oe_subject
+entity_type: node
+bundle: oe_consultation
+label: Subject
+description: ''
+required: true
+translatable: true
```
Comment (yenyasinn) [inline modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_subject.yml:16]: `translatable: false`

File: modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_summary.yml [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+  module:
+    - text
+id: node.oe_consultation.oe_summary
+field_name: oe_summary
+entity_type: node
+bundle: oe_consultation
+label: Introduction
+description: ''
+required: true
+translatable: true
```
Comment (yenyasinn) [inline modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_summary.yml:16]: `translatable: false`

File: modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_teaser.yml [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+  module:
+    - text
+id: node.oe_consultation.oe_teaser
+field_name: oe_teaser
+entity_type: node
+bundle: oe_consultation
+label: Teaser
+description: ''
+required: true
+translatable: true
```
Comment (yenyasinn) [inline modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_teaser.yml:17]: `translatable: false`

File: modules/oe_content_consultation/config/install/field.storage.node.oe_consultation_contacts.yml [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+id: node.oe_consultation_contacts
+field_name: oe_consultation_contacts
+entity_type: node
+type: entity_reference
+settings:
+  target_type: oe_contact
+module: core
+locked: false
+cardinality: -1
+translatable: true
```
Comment (yenyasinn) [inline modules/oe_content_consultation/config/install/field.storage.node.oe_consultation_contacts.yml:16]: `translatable: false`

File: modules/oe_content_consultation/config/install/field.storage.node.oe_consultation_deadline.yml [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+id: node.oe_consultation_deadline
+field_name: oe_consultation_deadline
+entity_type: node
+type: datetime
+settings:
+  datetime_type: datetime
+module: datetime
+locked: false
+cardinality: 1
+translatable: true
```
Comment (yenyasinn) [inline modules/oe_content_consultation/config/install/field.storage.node.oe_consultation_deadline.yml:16]: `translatable: false` as in Call for proposals

File: modules/oe_content_consultation/config/install/field.storage.node.oe_consultation_opening_date.yml [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+id: node.oe_consultation_opening_date
+field_name: oe_consultation_opening_date
+entity_type: node
+type: datetime
+settings:
+  datetime_type: date
+module: datetime
+locked: false
+cardinality: 1
+translatable: true
```
Comment (yenyasinn) [inline modules/oe_content_consultation/config/install/field.storage.node.oe_consultation_opening_date.yml:16]: `translatable: false` as in Call for proposals. Dates usually are untranslatable.

File: modules/oe_content_consultation/oe_content_consultation.info.yml [hunk 16]
```diff
@@ -0,0 +1,51 @@
+name: OpenEuropa Consultation Content
+description: Contains the EC corporate Consultation content type.
+package: OpenEuropa
+type: module
+core: 8.x
+
+dependencies:
+  - drupal:composite_reference
```
Comment (yenyasinn) [inline modules/oe_content_consultation/oe_content_consultation.info.yml:8]: `composite_reference:composite_reference`

**Review by sergepavle — CHANGES_REQUESTED** [review]
There are a few comments.

**Review by sergepavle — APPROVED** [review]

**Review by georgebaev — DISMISSED** [review]

**Review by sergepavle — APPROVED** [review]

File: modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_summary.yml [hunk 17]
```diff
@@ -0,0 +1,20 @@
...
+    - node.type.oe_consultation
+  module:
+    - text
+id: node.oe_consultation.oe_summary
+field_name: oe_summary
+entity_type: node
+bundle: oe_consultation
+label: Introduction
+description: ''
+required: true
```
Comment (upchuk) [inline modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_summary.yml:15]: Why is this required?

File: modules/oe_content_consultation/config/install/core.entity_form_display.node.oe_consultation.default.yml [hunk 18]
```diff
@@ -0,0 +1,319 @@
...
+  oe_summary:
+    weight: 1
+    settings:
+      rows: 5
+      placeholder: ''
+    third_party_settings:
+      maxlength:
+        maxlength_js: 250
+        maxlength_js_label: 'Content limited to @limit characters, remaining: <strong>@remaining</strong>'
+        maxlength_js_enforce: false
```
Comment (upchuk) [inline modules/oe_content_consultation/config/install/core.entity_form_display.node.oe_consultation.default.yml:254]: Shouldn't these 2 be set to TRUE (compared to the rest of the CTs)?

File: modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_summary.yml [hunk 19]
```diff
@@ -0,0 +1,20 @@
...
+    - field.storage.node.oe_summary
+    - node.type.oe_consultation
+  module:
+    - text
+id: node.oe_consultation.oe_summary
+field_name: oe_summary
+entity_type: node
+bundle: oe_consultation
+label: Introduction
+description: ''
```
Comment (upchuk) [inline modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_summary.yml:14]: We are missing the description. See the use of this AND all the other common fields and make them consistent with the rest of the CTs.

File: modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_author.yml [hunk 20]
```diff
@@ -0,0 +1,34 @@
+langcode: en
```
Comment (upchuk) [inline modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_author.yml:1]: It seems there is a mistake in the ticket, not finding `Author` neither in the common fields nor in the consulttion reqs.

**Review by upchuk — DISMISSED** [review]
Review is not done yet, passing to @ademarco

File: modules/oe_content_consultation/tests/src/Kernel/ConsultationNodeWrapperTest.php [hunk 21]
```diff
@@ -0,0 +1,221 @@
...
+use Drupal\oe_content_consultation\ConsultationNodeWrapper;
+use Drupal\oe_content\CallEntityWrapperInterface;
+
+/**
+ * Tests "Consultation" wrapper class.
+ *
+ * @coversDefaultClass \Drupal\oe_content_consultation\ConsultationNodeWrapper
+ * @group oe_content_consultation
+ */
+class ConsultationNodeWrapperTest extends ConsultationKernelTestBase {
```
Comment (yenyasinn) [inline modules/oe_content_consultation/tests/src/Kernel/ConsultationNodeWrapperTest.php:21]: There is code duplication in ConsultationNodeWrapperTest, CallForProposalsNodeWrapperTest, EventNodeWrapperTest, CallForTendersNodeWrapperTest. Might be trait for node wrappers with methods `testUnsupportedEntityType()`, `testUnsupportedEntityBundle()`, `testGetters()`, `assertGetters` should be created?
@nagyad

Comment (nagyad) [reply modules/oe_content_consultation/tests/src/Kernel/ConsultationNodeWrapperTest.php:21]: Why not. Let's open a followup ticket for this.

File: tests/features/consultation.feature [hunk 22]
```diff
@@ -0,0 +1,77 @@
...
+    And I fill in "Legal notice" with "Legal notice text"
+
+    # Create General contact.
+    And I press "Add new Contact"
+    And I wait for AJAX to finish
+    And I fill in "Name" with "General contact"
+    And I fill in "Organisation" with "General contact organisation"
+    And I fill in "Body text" with "General contact body text"
+    And I fill in "Website" with "https://general-contact.com"
+    And I press "Create Contact"
```
[the remark this answers is not in the document]
Comment (yenyasinn) [reply tests/features/consultation.feature:38]: I mean we need to test that contact is shown on the page after saving. See https://github.com/openeuropa/oe_content/blob/master/tests/features/call_proposals.feature#L104

File: tests/features/consultation.feature [hunk 23]
```diff
@@ -0,0 +1,78 @@
...
+    And I fill in "Consultation outcome" with "Consultation outcome text"
+    And I fill in "Use existing media" with "My Document 1"
+    And I fill in "Additional information" with "Additional information text"
+    And I fill in "Legal notice" with "Legal notice text"
+
+    # Create General contact.
+    And I press "Add new Contact"
+    And I wait for AJAX to finish
+    And I fill in "Name" with "General contact"
+    And I fill in "Organisation" with "General contact organisation"
```
Comment (yenyasinn) [inline tests/features/consultation.feature:35]: Why do we need to fill optional fields to the Contact entity if we don't check them after? I think we can remove them, or need to check that they exist on the page.

**Review by yenyasinn — APPROVED** [review]

File: modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_consultation_deadline.yml [hunk 24]
```diff
@@ -0,0 +1,20 @@
...
+  config:
+    - field.storage.node.oe_consultation_deadline
+    - node.type.oe_consultation
+  module:
+    - datetime
+id: node.oe_consultation.oe_consultation_deadline
+field_name: oe_consultation_deadline
+entity_type: node
+bundle: oe_consultation
+label: 'Deadline date'
```
Comment (yenyasinn) [inline modules/oe_content_consultation/config/install/field.field.node.oe_consultation.oe_consultation_deadline.yml:13]: Based on spec https://webgate.ec.europa.eu/fpfis/wikis/pages/viewpage.action?spaceKey=EWPP&title=Consultation+Content+Type+Specification+-+Version+1.0.0 it has to be "Deadline"

**Review by sergepavle — CHANGES_REQUESTED** [review]
Little comment. Also Deadline date should be renamed to Deadline.

**Comment by yenyasinn** [comment]
> I've left little comment. Also, `Deadline date` should be renamed to `Deadline`.

This is unclear for now.

**Review by yenyasinn — APPROVED** [review]

**Review by sergepavle — APPROVED** [review]
